### PR TITLE
Refactor process package and introduce ProcessManager{} with tests

### PR DESCRIPTION
**Pull request by metalmatze**
This is still WIP and other files need to be updated.
Some funcs that use this need get a ProcessManager instance passed.

### CODE REVIEW

File: modules/process/manager.go [hunk 1]
```diff
@@ -9,52 +9,71 @@ import (
...
+// Exec a command and use a specific timeout duration.
+func (pm *ProcessManager) ExecTimeout(timeout time.Duration, desc, cmdName string, args ...string) error {
+	return pm.ExecDir(timeout, "", desc, cmdName, args...)
+}
+
+// Exec a command with a timeout in given dir.
+func (pm *ProcessManager) ExecDir(timeout time.Duration, dir, desc, cmdName string, args ...string) error {
 	if timeout == -1 {
-		timeout = DEFAULT
+		timeout = DefaultTimeout
```
Comment (tboerger) [inline modules/process/manager.go]: Just set the default within the exec function?

**Comment by metalmatze** [comment]
This is ready to merge.

File: modules/process/manager.go [hunk 2]
```diff
@@ -9,66 +9,98 @@ import (
...
+		manager = &Manager{
+			Processes: make(map[int64]*Process),
+		}
+	}
+	return manager
+}
+
+// Add a process to the ProcessManager and returns its PID.
+func (pm *Manager) Add(description string, cmd *exec.Cmd) int64 {
+	pid := pm.counter + 1
```
Comment (lunny) [inline modules/process/manager.go]: Should this line put after the lock line?

**Comment by tboerger** [comment]
@metalmatze travis is failing.

**Comment by metalmatze** [comment]
Travis fails because there's no openssh-client on travis. I think I'll leave this for now, until we're drone only. Then this will pass.

**Comment by strk** [comment]
@metalmatze did you try adding `openssh-client` to the `apt-get install` line in `before_install` of .travis.yml ?

**Comment by lunny** [comment]
This PR is waiting until drone is ready. @tboerger

**Comment by strk** [comment]
Drone PR is #96

**Comment by lunny** [comment]
what's the block?

**Comment by tboerger** [comment]
I have dropped the blocked panel

**Comment by lunny** [comment]
So this only needed to rebase

**Comment by tboerger** [comment]
Yes, and @metalmatze said that he will finish it this weekend

**Comment by tboerger** [comment]
Do we want to move it to 1.1 @metalmatze?

**Comment by lunny** [comment]
I just move this to 1.1 and if @metalmatze back and 1.0 is not released yet, please move it back.

**Comment by lunny** [comment]
Please rebase and we can review continue.

**Comment by tboerger** [comment]
@metalmatze can you finish that and resolve the conflicts?

**Comment by metalmatze** [comment]
Yes I want to, but currently at 33C3 there's to less time. I'll try to rebase and finish on the weekend.

**Comment by metalmatze** [comment]
Ok. Instead of rebasing I've now copied over my changes in `modules/process` and simply refactored all its use cases.

**Comment by lunny** [comment]
Is this ready for review again?

**Comment by metalmatze** [comment]
I guess. I fixed all the errors and rebased to the current `master`.

File: modules/process/manager.go (hunk 2 above)
Comment (lunny) [reply modules/process/manager.go]: ^

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
otherwise LGTM

File: modules/process/manager.go (hunk 2 above)
Comment (metalmatze) [reply modules/process/manager.go]: Not necessarily. But probably better.

**Review by lunny — APPROVED** [review]

**Comment by tboerger** [comment]
LGTM

**Pull request merged**